src/param/int-list.js

```diff
@@ -0,0 +1,34 @@
+const assert = require('assert');
+const List = require('./list');
+
+class IntList extends List {
```

Comment:
You should extend `NumberList` here not `List`. `NumberList` already has this validation in it. We don't want to duplicate the code here :)